src/tscore/unit_tests/test_Extendible.cc

```diff
@@ -209,20 +209,36 @@ TEST_CASE("Extendible", "")
     CHECK(testField::alive == 0);
   }
 
+  INFO("AcidPtr AcidCommitPtr malloc ptr int");
+  {
+    void *mem            = malloc(sizeof(AcidPtr<int>));
+    AcidPtr<int> &reader = *(new (mem) AcidPtr<int>);
+    {
+      auto writer = reader.startCommit();
+      CHECK(*writer == 0);
+      *writer = 1;
+      CHECK(*writer == 1);
+      CHECK(*reader.getPtr().get() == 0);
+      // end of scope writer, commit to reader
+    }
+    CHECK(*reader.getPtr().get() == 1);
+    reader.~AcidPtr<int>();
+    free(mem);
+  }
   INFO("AcidPtr AcidCommitPtr casting");
   {
-    void *mem = malloc(sizeof(AcidPtr<testField>));
-    new (mem) AcidPtr<testField>();
-    AcidPtr<testField> &reader = *static_cast<AcidPtr<testField> *>(mem);
+    void *mem                  = malloc(sizeof(AcidPtr<testField>));
+    AcidPtr<testField> &reader = *(new (mem) AcidPtr<testField>);
```

Comment:
Why aren't you calling new and delete and letting new allocate the memory?

Reply:
Do you mean like this?  https://stackoverflow.com/questions/4145595/why-call-operator-new-explicitly

Does ASAN get unhappy when we call malloc() and free() rather than operator new and delete?

Reply:
asan can handle malloc/free and new/delete.  I was wondering why he wasn't using new and delete, but this was explain in our standup.